src/main/java/core/basesyntax/service/ShopServiceImpl.java

```diff
@@ -0,0 +1,37 @@
+package core.basesyntax.service;
+
+import core.basesyntax.dao.FruitDao;
+import core.basesyntax.domain.Fruit;
+import core.basesyntax.service.strategy.OperationStrategy;
+import java.util.List;
+
+public class ShopServiceImpl implements ShopService {
+    private static final String APPLE_NAME = "apple";
+    private static final String BANANA_NAME = "banana";
+    private final OperationStrategy operationStrategy;
+    private final FruitDao fruitDao;
+
+    public ShopServiceImpl(OperationStrategy operationStrategy, FruitDao fruitDao) {
+        this.operationStrategy = operationStrategy;
+        this.fruitDao = fruitDao;
+    }
+
+    @Override
+    public void process(List<Fruit> fruits) {
+        int appleQuantity = 0;
+        int bananaQuantity = 0;
+        for (Fruit currentFruit : fruits) {
+            Fruit.Operation currentOperation = currentFruit.getOperation();
+            int currentQuantity = currentFruit.getQuantity();
+            if (currentFruit.getName().equals(APPLE_NAME)) {
+                appleQuantity = operationStrategy.get(currentOperation)
+                        .getQuantity(appleQuantity, currentQuantity);
+            } else if (currentFruit.getName().equals(BANANA_NAME)) {
+                bananaQuantity = operationStrategy.get(currentOperation)
+                        .getQuantity(bananaQuantity, currentQuantity);
+            }
+        }
+        fruitDao.add(new Fruit(APPLE_NAME, appleQuantity));
```

Comment:
The 'add' method of 'fruitDao' is being called with hardcoded fruit names, which makes it difficult to extend the code for other types of fruits. Consider using a more dynamic approach that doesn't rely on hardcoded values.